events-api/src/modules/events/events.controller.ts

```diff
@@ -23,3 +29,64 @@ EventsController.get(
     });
   }
 );
+
+EventsController.post(
+  "/",
+  validateRequestBody(eventsCreateDtoSchema),
+  async (req, res) => {
+    await EventsService.createEvent(req.body);
+
+    return res.status(201).json({
+      message: "Event created successfully",
+    });
+  }
+);
+
+EventsController.patch(
+  "/:id",
+  validateRouteParams(eventsRouteParamsDtoSchema),
+  validateRequestBody(eventsUpdateDtoSchema),
+  async (req, res) => {
+    const event = await EventsService.updateEvent(
+      req.body,
+      req.params["id"] as string
+    );
+    if (event === null) {
+      return res.status(404).json({ message: "Event not found" });
+    }
+
+    return res.status(201).json({
+      message: "Event updated successfully",
+    });
+  }
+);
+
+EventsController.delete(
+  "/:id",
+  validateRouteParams(eventsRouteParamsDtoSchema),
+  async (req, res) => {
+    const event = await EventsService.deleteEvent(req.params["id"] as string);
+    if (event === null) {
+      return res.status(404).json({ message: "Event not found" });
+    }
+
+    return res.status(204).json({
+      message: "Event deleted successfully",
+    });
+  }
+);
+
+EventsController.get(
+  "/:id/tickets",
+  validateRouteParams(eventsRouteParamsDtoSchema),
+  async (req, res) => {
+    const tickets = await TicketsService.getTicketsForEvent(
+      req.params["id"] as string
```

Comment:
Don't use casting, better to move the retrieval of id param to the some utility function

Reply:
done

Reply:
Replace the casting with utility function, the casting is still here: https://github.com/NegatiV-64/expressjs-masterclass-2024/pull/2/files#diff-45c13e19a9dcdf712910a3e733f33a51eb0c85992329ff8034d9a1b98f0b44c7R88